app/src/main/java/com/azuresamples/azureadsampleapp/MainActivity.java

```diff
@@ -234,6 +270,52 @@ private void onSignOutClicked() {
     // updateSignedOutUI() - Updates UI when app sign out succeeds
     //
 
+    private String getClaims(String authenticateHeaderValue){
+
+        String instructionDelimiter = "\\s*,\\s*";
+        String claimsValue = null;
+
+        if(authenticateHeaderValue != null) {
+            String[] instructions = authenticateHeaderValue.split(instructionDelimiter);
+
+            for (String instruction : instructions) {
+                if (instruction.toLowerCase().startsWith("claims")) {
+                    claimsValue = instruction.substring(instruction.indexOf("=") + 2, instruction.length() - 1);
```

Comment:
suggestion: escape here instead of going through the remaining instructions (potentially saving a bit of cpu time). should there be more claims instructions, then right now the last one wins, and with the other one the first one "wins"